hawc/apps/epiv2/mixins.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import exceptions, serializers
+
+from .models import Design
+
+
+class BelongsToSameDesignMixin:
+    """
+    ex: when updating an ExposureLevel, there's a related Chemical and Exposure sometimes passed into the payload as
+    "chemical_id" and "exposure_level_id". Basic DRF functionality verifies that the supplied values are valid
+    items in the database, but it's a common requirement that they also belong to the same parent object; in this
+    case, the same design. This mixin performs that type of validation logic.
+
+    Could potentially generalize this even further (so that it's not just useful for "Design"); this type of
+    relationship is not uncommon.
+    """
+
```

Comment:
add a type annotation here: `belongs_to_same_design_fields: list[tuple[str, type[models.Model]]`

Reply:
added type annotation in the mixin, e.g. 

```
class SameDesignSerializerMixin:
   def validate(self, data):
      fields_to_check: list[tuple[str, type[models.Model]]] = self.belongs_to_same_design_fields
```

Reply:
I was thinking more like this?

```python
class BelongsToSameDesignMixin:
    """
    ex: when updating an ExposureLevel, there's a related Chemical and Exposure sometimes passed into the payload as
    "chemical_id" and "exposure_level_id". Basic DRF functionality verifies that the supplied values are valid
    items in the database, but it's a common requirement that they also belong to the same parent object; in this
    case, the same design. This mixin performs that type of validation logic.

    Could potentially generalize this even further (so that it's not just useful for "Design"); this type of
    relationship is not uncommon.
    """ 

    fields_to_check: list[tuple[str, type[models.Model]]]

    def validate(self, data):
        ...

```